src/fragmentarium/ui/search/NumberSearchForm.tsx

```diff
@@ -1,57 +1,33 @@
 import React, { Component } from 'react'
-import { stringify } from 'query-string'
-import { Form, Button, Row, Col } from 'react-bootstrap'
-import { withRouter, RouteComponentProps } from 'react-router-dom'
-import { History } from 'history'
+import { Col, Form, Row } from 'react-bootstrap'
 
 type Props = {
-  number: string | null | undefined
-  history: History
-} & RouteComponentProps
-
-class FragmentSearchForm extends Component<Props> {
-  state = {
-    number: this.props.number || '',
-  }
+  onChange(searchForm: string, searchQuery: string): void
+  getState(key: string): string
+}
 
+class NumberSearchForm extends Component<Props> {
   onChange = (event) => {
-    this.setState({
-      number: event.target.value,
-    })
-  }
-
-  submit = (event) => {
-    event.preventDefault()
-    this.props.history.push(
-      `/fragmentarium/search/?${stringify({
-        number: this.state.number,
-      })}`
-    )
+    this.props.onChange('number', event.target.value || '')
```

Comment:
You should only pass the value and not couple the implementation with something on the outside. Whoever defines the callback should make do the correct thing without the form knowing its internals.